app/battle/battle_service.py

```diff
@@ -0,0 +1,18 @@
+from app.knights_person.knight import Knight
+
+
+class Battle:
+    @staticmethod
+    def fight(first: Knight, second: Knight) -> None:
+        first.hp -= second.power - first.protection
+        second.hp -= first.power - second.protection
+
+        if first.hp < 0:
+            first.hp = 0
+
+        if second.hp < 0:
+            second.hp = 0
+
+    @staticmethod
+    def results(knights: list) -> dict:
```

Comment:
add additional typehint for list inner objects like list[YOUR_VALUE_TYPE]
for example: list that contains string elements has typehint list[str]